src/main/java/duke/util/FormatChecker.java

```diff
@@ -0,0 +1,159 @@
+            throw new InvalidFormatDeleteException();
+        }
+    }
+
+    /**
+     * Checks if the done command has the description.
+     *
+     * @param inputArr String array representing the command the user keyed in.
+     * @throws InvalidFormatDoneException When there are missing numbers or Not a number behind delete.
+     */
+    public static void checkDoneFormat(String ... inputArr) throws InvalidFormatDoneException {
+        if (inputArr.length == 1 || !isNumber(inputArr[1])) {
+            throw new InvalidFormatDoneException();
+        }
+    }
+
+    /**
+     * Checks if the find command has the description.
+     *
+     * @param inputArr String array representing the command the user keyed in.
+     * @throws InvalidFormatFindException When there are missing description or user tried to key in more than
+     * one keyword.
+     */
+    public static void checkFindFormat(String ... inputArr) throws InvalidFormatFindException {
+        if (inputArr.length == 1) {
+            throw new InvalidFormatFindException();
+        }
+        String[] inputArr2 = inputArr[1].split(KEYWORD_ONE_SPACE);
+        if (inputArr2.length > 1) {
+            throw new InvalidFormatFindException();
+        }
+    }
+
+    /**
+     * Checks if the help command has the description.
+     *
+     * @param inputArr String array representing the command the user keyed in.
+     * @throws InvalidFormatHelpException When there are extra command behind help.
+     */
+    public static void checkHelpFormat(String ... inputArr) throws InvalidFormatHelpException {
+        if (inputArr.length != 1) {
+            throw new InvalidFormatHelpException();
+        }
+    }
+
+    /**
+     * Checks if the list command has the description.
+     *
+     * @param inputArr String array representing the command the user keyed in.
+     * @throws InvalidFormatListException When there are extra command behind list.
+     */
+    public static void checkListFormat(String ... inputArr) throws InvalidFormatListException {
+        if (inputArr.length != 1) {
+            throw new InvalidFormatListException();
+        }
+    }
+
+    /**
+     * Checks if the remind command has the description.
+     *
+     * @param inputArr String array representing the command the user keyed in.
+     * @throws InvalidFormatReminderException When there are missing description or if user key in more than the
+     * required format.
+     */
+    public static void checkReminderFormat(TaskList tasks, String ... inputArr) throws InvalidFormatReminderException {
+        if (inputArr.length == 1) {
+            throw new InvalidFormatReminderException();
+        }
+        String[] inputArr3 = inputArr[1].split(KEYWORD_ONE_SPACE);
+        if (inputArr3.length != 2) {
+            throw new InvalidFormatReminderException();
+        }
+        if (!inputArr3[1].toLowerCase().equals("y") && !inputArr3[1].toLowerCase().equals("n")) {
```

Comment:
Boolean seems a bit long here. I think it might benefit to abstract this if condition out.

Reply:
thanks